radixdlt/src/main/java/com/radixdlt/consensus/bft/VertexStore.java

```diff
@@ -66,36 +65,52 @@
 	private QuorumCertificate highestQC;
 	private QuorumCertificate highestCommittedQC;
 
-	public VertexStore(
+	private VertexStore(
 		VerifiedVertex rootVertex,
 		QuorumCertificate rootQC,
 		Ledger ledger,
 		BFTUpdateSender bftUpdateSender,
 		VertexStoreEventSender vertexStoreEventSender,
 		SystemCounters counters
 	) {
-		this(rootVertex, rootQC, Collections.emptyList(), ledger, bftUpdateSender, vertexStoreEventSender, counters);
+		this.ledger = Objects.requireNonNull(ledger);
+		this.vertexStoreEventSender = Objects.requireNonNull(vertexStoreEventSender);
+		this.bftUpdateSender = Objects.requireNonNull(bftUpdateSender);
+		this.counters = Objects.requireNonNull(counters);
+		this.rootVertex = Objects.requireNonNull(rootVertex);
+		this.highestQC = Objects.requireNonNull(rootQC);
```

Comment:
Checks above may throw exception, so they should be shifted to factory method and this constructor then will be just a plain all-args constructor with no chances to throw exception.